### DISCUSSION TITLE
2.0.3 beta

### DISCUSSION THREAD

**Discussion by camueller** [camueller/SmartApplianceEnabler#233] [Releases]
Details zu den Änderungen finden sich im [Changelog](https://github.com/camueller/SmartApplianceEnabler/blob/2.0/CHANGELOG.md#203---20032022),

:exclamation: Die [Dokumentation zu SAE 2.x](https://github.com/camueller/SmartApplianceEnabler/tree/2.0) befindet sich auf einem Git-Branch, d.h. wenn man "normal" auf die Github-Seite geht, sieht man die Dokumentation für SAE 1.x.  Deshalb entweder den Link zu Beginn des Kommentars anklicken, oder den selbst den Branch "2.0" auswählen.

**Comment by ul-gh** [comment]
Hi,


SAE ist für mich neu, habe nun als Einstieg in das System aber zuerst die Version 2.0.3 beta installiert.

Installiert habe ich auf einem Intel-kompatiblen 32-Bit-Serversystem, das als NAS, Datenlogger, Node-Red-Server und als MQTT-Broker bereits seit vielen Jahren im Dauerbetrieb läuft - die GPIOs brauche ich erstmal nicht, bzw. das wird später über eine per HTTP steuerbare Relaiskarte o.ä. nachgerüstet.


Was ich berichten kann von der Installation der 2.0.3 Beta:

=> Den Ort der Einstellungsdatei zum Ändern des Standard-Ports des SAE (8080) musste ich suchen, das ist die /etc/default/smartapplianceenabler - wäre nett, wenn das in die Dokumentation kommt (falls ich das nicht überlesen haben sollte)

(Der Port 8080 ist auf meinem System bereits vom einem anderen Dienst belegt, den ich nicht umstellen kann)

=> Der Node-Red-Flow für den eingerichteten Tasmota-Smart-Plug ließ sich importieren; Zwei Abhängigkeiten musste ich über die Node-Red-Palette noch nachinstallieren: node-red-contrib-ui-timelines-chart und node-red-node-ui-table.

=> Node-Red lieferte im Debug-Log weitere Fehler: "Cannot read property of undefined",  "datefns", "no such property": Hier habe ich das Node-Modul "date-fns" im Basisordner der Node-Red-Installation (bei mir ~/.node-red/) installiert (die Option --save ist ab NPM 5.0 dabei obsolet):
```
cd ~/.node-red
npm i date-fns
```
und dort im selben Ordner in der Datei settings.js dem globalen Kontext der Node-Red-Instanz hinzugefügt:
```
    functionGlobalContext: {
        datefns: require('date-fns')
    },
```

Soweit lief die Installation scheinbar durch.

Eine über HTTP angebundene Tasmota-Steckdose wurde auch auf Anhieb im Sunny Portal gefunden.

Ich musste rätselraten, wie ich die korrekte Einstellung der HTTP-URL und der GET-Requests prüfen kann (die Schaltfunktion lässt ich testen, in dem auf das rote oder grüne Ampelsymbol geklickt wird, das habe ich herausgefunden).

Im ersten Test mit Solarertrag heute wurde die Last scheinbar auch geschaltet, wobei im Sunnyportal nur ein Teil der möglichen Zeit auch geloggt wurde. Woran das liegt weiß ich nicht, den eingestellten Zeitplan bekomme ich im Sunnyportal nicht angezeigt.

Wohl angezeigt wird der eingestellte Zeitplan im Timelines-Chart im Node-Red, wobei mir auffällt, dass dort nur der gesamte Plan als "optional" angezeit wird, obwohl eine Mindestlaufzeit von 12 Stunden eingerichtet ist.
(Siehe Dateianhang Screenshot mit Zeitplan)

=> Der Plot der Leistung über die Zeit ist so skaliert, dass die X-Achse nur 10 Minuten umfasst. Das ist IMO zu kurz, um nützlich zu sein.

![timeframes_SAE](https://user-images.githubusercontent.com/16521333/159984722-a24b0085-adb9-4af0-9fca-a894e6de022e.png)

**Reply by camueller** [reply]
Hinweise zur `/etc/default/smartapplianceenabler` finden sich in https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/ConfigurationFiles_DE.md, die von mehreren Stellen verlinkt ist, u.a. von https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/ManualInstallation_DE.md . Wo hättest Du denn diese Links sonst erwartet?

Die Installation von NodeRED bzw. dafür benötige Bibliotheken ist ebenfalls beschrieben: https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/NodeRED_DE.md

Auch die Funktion der Ampel ist beschrieben: https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/Status_DE.md

Warum der Zeitplan bei Dir als optional dargestellt wird, muss ich prüfen. Er ist bei Dir teilweise optional, weil Du eine Mindestlaufzeit von 12:00 Stunden und ein Maximale Laufzeit von 12:05 angegeben hast, d.h. 5 Minuten sind optional. In der Praxis macht so eine Vorgaben keinen Sinn.

Der Leistungsplot ist deshalb so kurz, weil er primar zur Erkennung der Leistung für Geräte mit Anlaufstromerkennung gedacht ist. Es soll kein Ersatz für das Sunny Portal sein. Aber diese Wert kann man ja in NodeRED nach Bedarf anpassen :-)

**Comment by ul-gh** [comment]
Hi,

vielen Dank für die Antwort und natürlich für die Software per se.

Mein Posting  war nur als Erfahrungsbericht gedacht, das ist so wie ich die erste Installation erlebt habe.

Ja, die Konfiguration ist in den einzelnen Artikeln sehr gut beschrieben - ich habe nur die [manuelle Installationsanleitung](https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/ManualInstallation_DE.md) abgearbeitet, die ist schon sehr detailliert, insbesondere was die Vorbereitung des Raspberry Pi betrifft - aber natürlich kann da nicht jede Eventualität drin stehen.

Für die Version 2.0 könnte/müsste/sollte natürlich dort in der Installationsanleitung ein Link auf die [Beschreibung zur Node-Red-Installation ](https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/NodeRED_DE.md)und den Abhängigkeiten gesetzt werden, das wäre das i-Tüpfelchen zur Vollständigkeit.

Wenn gewünscht, werde ich auch aktive Rückmeldung geben, wenn etwas getestet werden soll.
Kann jeweils ein paar Tage dauern, weil die Anlage zu Hause ist, und ich nur an den Wochenenden dort bin, bzw. werktags unterwegs auf Montage.

Viele Grüße,
Ulrich

**Reply by camueller** [reply]
Danke für den Hinweis zum fehlenden Link zur Node-RED-Installation. Habe ihn in die [manuelle Installationsanleitung](https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/ManualInstallation_DE.md) eingefügt.

**Comment by Tonno87** [comment]
Hi Zusammen,

ich habe mit Version 2.0.1 gestartet und habe einen Heizlüfter den ich gerne über eine Tasmota Steckdose immer dann Heizen lassen will wenn genug Überschuss da ist. Das klappt auch soweit immer am ersten Tag nachdem ich den SAE gestartet habe. Ab dem zweiten Tag schaltet die Steckdose nicht mehr. 
Sobald der SAE neu gestartet wird läuft sofort wieder alles bis zum Folgetag. Ich bin etwas ratlos, vielleicht hat jemand eine Idee was falsch eingestellt ist?!

Laut dem Sunny Portal wird der Heizlüfter auf jeden Fall eingeplant. 

![Screenshot_20220330-160023~2](https://user-images.githubusercontent.com/59800400/160857514-17330f2a-2597-4075-b89a-38728779b3f2.png)
![Screenshot_20220330-160032~2](https://user-images.githubusercontent.com/59800400/160857522-57631bbd-89aa-4e83-8bc9-76cf9dca5e9d.png)
![Screenshot_20220330-160601](https://user-images.githubusercontent.com/59800400/160857526-4fdd39b7-5583-424d-baa4-42e7d1d99ec9.png)
![Screenshot_20220330-160102](https://user-images.githubusercontent.com/59800400/160857528-d72a2cb3-590f-4890-bfd7-bf9d69c7fd17.png)

**Reply by camueller** [reply]
Dieser Thread handelt explizit von Version 2.0.3. Wenn Du also noch die 2.0.1 nutzt, solltest Du zunächst auf die 2.0.3 wechseln in der Hoffnung, dass das Problem da bereits behoben ist.

**Reply by Tonno87** [reply]
Okay um das Missverständnis direkt auszuräumen: Aktuell nutze ich 2.0.3, habe nur mit 2.0.1 angefangen.
Geht aber trotzdem nicht...

**Reply by camueller** [reply]
Leider fehlt der Anfang von Deinem ersten Screenshot, der genau für den Heizlüfter relevant ist. Das SEMP-XML lieber als formatierten Text posten, dann geht nix verloren.

**Reply by Tonno87** [reply]
Etwa so?

```
This XML file does not appear to have any style information associated with it. The document tree is shown below.
<Device2EM xmlns="http://www.sma.de/communication/schema/SEMP/v1">
	<script/>
	<DeviceInfo>
		<Identification>
			<DeviceId>F-0000000-000000000001-00</DeviceId>
			<DeviceName>Heizluefter</DeviceName>
			<DeviceType>Heater</DeviceType>
			<DeviceSerial>1</DeviceSerial>
			<DeviceVendor>Silvercrest</DeviceVendor>
		</Identification>
		<Characteristics>
			<MaxPowerConsumption>1200</MaxPowerConsumption>
			<MinOnTime>10</MinOnTime>
		</Characteristics>
		<Capabilities>
			<CurrentPower>
				<Method>Measurement</Method>
			</CurrentPower>
			<Timestamps>
				<AbsoluteTimestamps>false</AbsoluteTimestamps>
			</Timestamps>
			<Interruptions>
				<InterruptionsAllowed>true</InterruptionsAllowed>
			</Interruptions>
			<Requests>
				<OptionalEnergy>true</OptionalEnergy>
			</Requests>
		</Capabilities>
	</DeviceInfo>
	<DeviceInfo>
		<Identification>
			<DeviceId>F-0000000-000000000002-00</DeviceId>
			<DeviceName>EVC04</DeviceName>
			<DeviceType>EVCharger</DeviceType>
			<DeviceSerial>7000190521000181</DeviceSerial>
			<DeviceVendor>Vestel</DeviceVendor>
		</Identification>
		<Characteristics>
			<MaxPowerConsumption>11000</MaxPowerConsumption>
			<MinPowerConsumption>1380</MinPowerConsumption>
		</Characteristics>
		<Capabilities>
			<CurrentPower>
				<Method>Measurement</Method>
			</CurrentPower>
			<Timestamps>
				<AbsoluteTimestamps>false</AbsoluteTimestamps>
			</Timestamps>
			<Interruptions>
				<InterruptionsAllowed>true</InterruptionsAllowed>
			</Interruptions>
			<Requests>
				<OptionalEnergy>true</OptionalEnergy>
			</Requests>
		</Capabilities>
	</DeviceInfo>
	<DeviceInfo>
		<Identification>
			<DeviceId>F-0000000-000000000003-00</DeviceId>
			<DeviceName>Vitoladens 300-c</DeviceName>
			<DeviceType>Heater</DeviceType>
			<DeviceSerial>1</DeviceSerial>
			<DeviceVendor>Viessmamn</DeviceVendor>
		</Identification>
		<Characteristics>
			<MaxPowerConsumption>1000</MaxPowerConsumption>
		</Characteristics>
		<Capabilities>
			<CurrentPower>
				<Method>Estimation</Method>
			</CurrentPower>
			<Timestamps>
				<AbsoluteTimestamps>false</AbsoluteTimestamps>
			</Timestamps>
			<Interruptions>
				<InterruptionsAllowed>false</InterruptionsAllowed>
			</Interruptions>
			<Requests>
				<OptionalEnergy>false</OptionalEnergy>
			</Requests>
		</Capabilities>
	</DeviceInfo>
	<DeviceStatus>
		<DeviceId>F-0000000-000000000001-00</DeviceId>
		<EMSignalsAccepted>true</EMSignalsAccepted>
		<Status>Off</Status>
		<PowerConsumption>
			<PowerInfo>
				<AveragePower>0</AveragePower>
				<Timestamp>0</Timestamp>
				<AveragingInterval>60</AveragingInterval>
			</PowerInfo>
		</PowerConsumption>
	</DeviceStatus>
	<DeviceStatus>
		<DeviceId>F-0000000-000000000002-00</DeviceId>
		<EMSignalsAccepted>false</EMSignalsAccepted>
		<Status>Off</Status>
		<PowerConsumption>
			<PowerInfo>
				<AveragePower>0</AveragePower>
				<Timestamp>0</Timestamp>
				<AveragingInterval>60</AveragingInterval>
			</PowerInfo>
		</PowerConsumption>
	</DeviceStatus>
	<DeviceStatus>
		<DeviceId>F-0000000-000000000003-00</DeviceId>
		<EMSignalsAccepted>false</EMSignalsAccepted>
		<Status>On</Status>
		<PowerConsumption>
			<PowerInfo>
				<AveragePower>1000</AveragePower>
				<Timestamp>0</Timestamp>
				<AveragingInterval>60</AveragingInterval>
			</PowerInfo>
		</PowerConsumption>
	</DeviceStatus>
	<PlanningRequest>
		<Timeframe>
			<DeviceId>F-0000000-000000000001-00</DeviceId>
			<EarliestStart>0</EarliestStart>
			<LatestEnd>3057</LatestEnd>
			<MinRunningTime>0</MinRunningTime>
			<MaxRunningTime>3057</MaxRunningTime>
		</Timeframe>
		<Timeframe>
			<DeviceId>F-0000000-000000000001-00</DeviceId>
			<EarliestStart>31857</EarliestStart>
			<LatestEnd>89457</LatestEnd>
			<MinRunningTime>0</MinRunningTime>
			<MaxRunningTime>54000</MaxRunningTime>
		</Timeframe>
		<Timeframe>
			<DeviceId>F-0000000-000000000001-00</DeviceId>
			<EarliestStart>118257</EarliestStart>
			<LatestEnd>175857</LatestEnd>
			<MinRunningTime>0</MinRunningTime>
			<MaxRunningTime>54000</MaxRunningTime>
		</Timeframe>
	</PlanningRequest>
</Device2EM>
```

**Reply by camueller** [reply]
Alle relevanten Werte für den Heizlüfter sehen gut aus, d.h. der SHM sollte einen Einschaltbefehl schicken, wenn Überschuss vorhanden ist. Weiter oben hast Du geschrieben, dass es nach einem Neustart des SAE funktioniert, aber ab dem Folgetag nicht mehr. Vergleiche doch mal das obige XML direkt nach einer erfolgreichen Einschaltung durch den SHM mit dem XML am Tag danach.

**Reply by Tonno87** [reply]
Guten Abend,

also ich konnte die letzten Tage keine Unterschiede feststellen. In der Zwischenzeit habe ich noch Node Red installiert um das UI nutzen zu können. Was ich da festgestellt habe ist das sich das Gerät nicht über den Schalter aktivieren lässt. Oben in der Zeile steht wie am ersten Tag der Befehl vom Home Manager:
![image](https://user-images.githubusercontent.com/59800400/162808927-04c2da3a-01ce-43aa-88ce-1eb6357f6519.png)


Über die Tasmote-Website lässt sich die Steckdose jederzeit zuverlässig schalten.

Ich werde morgen mal die 1.x ausprobieren. Mal schauen ob's irgendwie bei mir am System liegt...

**Reply by Tonno87** [reply]
Also nun zum Ergebnis: 
Ich habe den lxc Container geklont, dann die beiden XML Datei gelöscht und die Version 1.6.20 heruntergeladen. Gestern Abend neu eingerichtet mit den selben Einstellungen wie bei 2.0.3. Das System selbst wurde also nicht verändert. Der Home Manager hat den Heizlüfter auch wieder erkannt.

Heizung läuft heute ohne Neustart so wie es sein sollte :)

**Reply by camueller** [reply]
> Was ich da festgestellt habe ist das sich das Gerät nicht über den Schalter aktivieren lässt.

Der Schalter ist nur zur Visualisierung des Schaltzutandes da. Ich wollte ursprünglich ein LED-Symbol verwenden, aber hatte keines gefunden, welches nicht das gesamte Layout kaputt gemacht hat.

**Reply by Tonno87** [reply]
Hm okay. Stellt sich die Frage ob ich noch mit Log-Dateien oder so dienen kann?! 
Ansonsten macht's für mich erstmal keinen Unterschied ob 1.6 oder 2.0 solange die Steckdosen jetzt schalten können. Daher würde ich bei dem jetzigen Zustand bleiben und das System weiter ausbauen

**Comment by geka79** [comment]
Hallo zusammen
Ich würde gerne den schon vorhanden MQTT Server verwenden, jedoch ist der mit User und PW geschützt.
Wo kann ich dies eintragen bzw. ist dies überhaupt möglich?

Danke

**Reply by Tonno87** [reply]
Hi,

der Eintrag **allow_anonymous true** muss in die conf. Datei von Mosquitto. Damit funktioniert es bei mir.

Gruß Dennis

**Reply by geka79** [reply]
hallo, ich würde gerne den MQTT vom iobroker verwenden, der hat jedoch user und PW.
Wo bzw. ob ich hier og. Kürzel eintragen kann muss ich erst herausfinden.

Danke für die schnelle Antwort :)

**Reply by Tonno87** [reply]
Ich nutze schon länger kein IOBroker mehr... Aber ich erinnere mich das man in den Einstellungen angeben konnte das unautorisierte Anfragen abgewiesen werden sollen.

**Reply by geka79** [reply]
mal schaun

**Reply by camueller** [reply]
Aktuell kan nder SAE keine MQTT-Broker verwenden, der Username/Passwort erfordert. Entweder Authentifzieren im MQTT-Broker deaktivieren oder separaten MQTT-Broker für SAE installieren und verwenden. Evtl. implementiere ich die Unterstützung von Authentifizierung am MQTT-Broker für die nächste SAE-Version - muss mal schauen, ob das einfach machbar ist ...

**Reply by camueller** [reply]
Authentifizierung mit Username/Passwort ist implementiert für die nächste Version.

**Reply by geka79** [reply]
Cool, danke
